Framework/MDAlgorithms/src/TransformMD.cpp

```diff
@@ -156,13 +156,35 @@ void TransformMD::exec() {
   if (histo) {
     // Recalculate all the values since the dimensions changed.
     histo->cacheValues();
-    if (m_scaling[0] < 0.0) {
-      signal_t *signals = histo->getSignalArray();
-      signal_t *errorsSq = histo->getErrorSquaredArray();
+    // Expect first 3 dimensions to be -1 for changing conventions
+    for (int i = 0; i < static_cast<int>(m_scaling.size()); i++)
+      if (m_scaling[i] < 0) {
+        std::vector<int> axes(m_scaling.size());        // vector with ints.
+        std::iota(std::begin(axes), std::end(axes), 0); // Fill with 0, 1, ...
+        axes[0] = i;
+        axes[i] = 0;
```

Comment:
If `i=0` the result of transposing is nothing? Can we start the loop at `1`, or am I missing something?

Reply:
If i=0 and m_scaling[0]<0, the axis is reversed.  This is always true when changing conventions.

Reply:
But wouldn't the resulting `axes` be something like `0, 1, 2, 3`, and `TransposeMD` is a no-op?

Reply:
transform needed for i=0

Reply:
You are right that the transpose does not need to be done for i=0, but the rest with the reverses